plugin.audio.somafm/resources/lib/addon.py

```diff
@@ -0,0 +1,191 @@
+import os
+import shutil
+import sys
+import urllib.request
+import urllib.parse
+import xml.etree.ElementTree as ET
+import time
+
+import xbmcaddon
+from xbmcgui import ListItem
+import xbmcplugin
+import xbmcgui
+import xbmc
+from xbmcplugin import SORT_METHOD_LISTENERS, SORT_METHOD_UNSORTED, SORT_METHOD_GENRE
+import xbmcvfs
+
+from resources.lib.channel import Channel
+
+CHANNELS_FILE_NAME = "channels.xml"
+
+__addon__ = "SomaFM"
+__addonid__ = "plugin.audio.somafm"
+__version__ = "2.0.0"
+
+__ms_per_day__ = 24 * 60 * 60 * 1000
+
+
+def log(msg):
+    xbmc.log(str(msg), level=xbmc.LOGDEBUG)
+
+
+log(sys.argv)
+
+rootURL = "https://somafm.com/"
+tempdir = xbmcvfs.translatePath("special://home/userdata/addon_data/%s" % __addonid__)
+xbmcvfs.mkdirs(tempdir)
+
+LOCAL_CHANNELS_FILE_PATH = os.path.join(tempdir, CHANNELS_FILE_NAME)
+
+try:
+    plugin_url = sys.argv[0]
+    handle = int(sys.argv[1])
+    query = sys.argv[2]
+except Exception as e:
+    xbmc.log("Initialization Failed: " + str(e), level=xbmc.LOGERROR)
```

Comment:
Python3 includes f strings so this can be changed to something like:

```suggestion
    xbmc.log(f "Initialization Failed: {e}", level=xbmc.LOGERROR)
```